### ISSUE TITLE
Join problem when using Skip and Take

### ISSUE THREAD

**Issue by AminEsmaeily** [aspnet/EntityFrameworkCore#14421]
I have a query like this:
``` c#
var data = context.Letters
	.Include(f => f.Referrals)
	.ThenInclude(f => f.Post)
	.Include(f => f.Attachments)
	.ThenInclude(f => f.Signs)
	.Include(f => f.Registerar)
	.Include(f => f.LetterChains)
	.Include(f => f.Secretariate)
	.Include(f => f.Zonecans)
	.ThenInclude(f => f.Zonecan)
	.Where(letter => !letter.Deleted && !letter.IsTemp);
```

This query returns a SQL command, something like following:
```sql
SELECT DISTINCT [r].[Id], ... , [r.Registerar].[UserName]
FROM [Letters] AS [r]
LEFT JOIN [Secretariates] AS [r.Secretariate] ON [r].[SecretariateId] = [r.Secretariate].[Id]
LEFT JOIN [AspNetUsers] AS [r.Registerar] ON [r].[RegisterarId] = [r.Registerar].[Id]
INNER JOIN (
    SELECT [f].*
    FROM [LetterToZonecans] AS [f]
    WHERE ([f].[Deleted] = 0) AND ([f].[ZonecanId] = 14)
) AS [t] ON [r].[Id] = [t].[LetterId]
WHERE (([r].[Deleted] = 0) AND ([r].[IsTemp] = 0)) AND (([r].[SecretariateId] = 1) AND ([r].[YearId] = 1))
ORDER BY [r].[RegisterDateTime] DESC, [r].[Id]
```

As you see, it joins to **AspNetUsers** and loads its columns. But when I add `Skip` and `Take` methods to the above  linq query, it doesn't join to the **AspNetUsers** table and finally, I get `null` values when I want to access to the **Registrar** properties of the Letter. Linq query with `Skip` and `Take` is as the following:
```c#
data = data.Distinct().Skip(paginationFrom).Take(paginationSize).OrderByDescending(f => f.RegisterDateTime);
```

SQL output of the above command:
``` sql
SELECT DISTINCT [t0].[Id], ... , [t0].[YearId]
FROM (
    SELECT [r].[Id], ... , [r].[YearId]
    FROM [Letters] AS [r]
    INNER JOIN (
        SELECT [f].*
        FROM [LetterToZonecans] AS [f]
        WHERE ([f].[Deleted] = 0) AND ([f].[ZonecanId] = 14)
    ) AS [t] ON [r].[Id] = [t].[LetterId]
    WHERE (([r].[Deleted] = 0) AND ([r].[IsTemp] = 0)) AND (([r].[SecretariateId] = 1) AND ([r].[YearId] = 1))
    ORDER BY (SELECT 1)
    OFFSET 0 ROWS FETCH NEXT 100 ROWS ONLY
) AS [t0]
ORDER BY [t0].[RegisterDateTime] DESC
```

Am I missing something or there is an issue with`Skip` and `Take`?

**Comment by AminEsmaeily** [comment]
Finally, I found the problem I fixed it. When I use `Distinct` method at the final part of my query, It doesn't join to **AspNetUsers**, but if I use it at the first part of my query, It works correctly. I should add `Distinct` to this part:
``` c#
var data = context.Letters
	.Include(f => f.Referrals)
	.ThenInclude(f => f.Post)
	.Include(f => f.Attachments)
	.ThenInclude(f => f.Signs)
	.Include(f => f.Registerar)
	.Include(f => f.LetterChains)
	.Include(f => f.Secretariate)
	.Include(f => f.Zonecans)
	.ThenInclude(f => f.Zonecan)
        .Distinct()
	.Where(letter => !letter.Deleted && !letter.IsTemp);
```

I don't understand why it matters to EF Core that when to use Distinct.

**Comment by smitpatel** [comment]
Looks like, Includes are getting ignored in 2nd query.

**Comment by AminEsmaeily** [comment]
Why the place of using the `Distinct` method on the main query can change the final generated SQL?

**Comment by smitpatel** [comment]
@AminEsmaeily - If Includes are getting ignored then of course generated SQL won't do joins to fetch related data. Use of `Distinct()` should not cause includes to be ignored. That is the bug.

**Comment by maumar** [comment]
blocked by #15611

**Comment by smitpatel** [comment]
@AminEsmaeily - We don't have enough code here to reproduce the error and see if it is fixed in latest release. Though in EF Core 3.1 we have redesigned whole query pipeline and believe that this issue should go away. I am closing this issue as fixed. If you still see the error after upgrading to EF Core 3.1, we will reopen and investigate.

**Comment by smitpatel** [comment]
@maumar - If you want to try repro this and add a regression test, feel free to reopen.
